bundler/lib/bundler/lockfile_parser.rb

```diff
@@ -111,6 +111,9 @@ def initialize(lockfile)
         elsif line == DEPENDENCIES
           @parse_method = :parse_dependency
         elsif line == CHECKSUMS
+          # This is a temporary solution to make this feature disabled by default
+          # for all gemfiles that don't already explicitly include the feature.
+          Bundler::Checksum.enable!
```

Comment:
this feels super brittle to me, given bundler could have read a different lockfile and then have stored the ivar.

I would feel a lot better if this got hoisted into the definition

Reply:
ok, I'll shift it there. Would you still use the same ivar approach? I toyed with adding a Bundler.setting for "enabled", but it makes it more complex. Do you prefer that?